vpr/src/place/place.cpp

```diff
@@ -448,8 +470,49 @@ void try_place(const t_placer_opts& placer_opts,
         }
     }
 
-    move_generator = std::make_unique<UniformMoveGenerator>();
+    int move_lim = 1;
+    move_lim = (int)(annealing_sched.inner_num * pow(cluster_ctx.clb_nlist.blocks().size(), 1.3333));
+
+    //create the move generator based on the chosen strategy
+    if (placer_opts.simpleRL_agent_placement == false) {
+        VTR_LOG("Using static probabilities for choosing each move type\n");
+        VTR_LOG("Probability of Uniform_move : %f \n", placer_opts.place_static_move_prob[0]);
+        VTR_LOG("Probability of Median_move : %f \n", placer_opts.place_static_move_prob[1]);
+        VTR_LOG("Probability of Weighted_median_move : %f \n", placer_opts.place_static_move_prob[2]);
+        VTR_LOG("Probability of Weighted_centroid_move : %f \n", placer_opts.place_static_move_prob[3]);
+        VTR_LOG("Probability of Timing_feasible_region_move : %f \n", placer_opts.place_static_move_prob[4]);
+        VTR_LOG("Probability of Critical_uniform_move : %f \n", placer_opts.place_static_move_prob[5]);
+        VTR_LOG("Probability of Centroid_move : %f \n", placer_opts.place_static_move_prob[6]);
+        move_generator = std::make_unique<StaticMoveGenerator>(placer_opts.place_static_move_prob);
+        move_generator2 = std::make_unique<StaticMoveGenerator>(placer_opts.place_static_move_prob);
+    } else {
+        if (placer_opts.place_agent_algorithm == E_GREEDY) {
+            VTR_LOG("Using simple RL 'Epsilon Greedy agent' for choosing move types\n");
+            std::unique_ptr<EpsilonGreedyAgent> karmed_bandit_agent1, karmed_bandit_agent2;
+            //agent's 1st state
+            karmed_bandit_agent1 = std::make_unique<EpsilonGreedyAgent>(4, placer_opts.place_agent_epsilon);
```

Comment:
What does the constant `4` mean?

Reply:
Is it the a list from 0 to this number of the moves types to allow?  If so, should definitely have a comment saying that, and noting what the first four move types are expected to be.

Same for number 7 below.

Reply:
Exactly as Vaughn said, 4 or 7 below are the number of the available actions in each state. All the actions are stored in a list starting from 0 and depending on the number of the available actions in each state, only a specific number of actions is used. I have added a comment to explain the different states of the agent and substituted 4 and 7 with constants.